pkg/controllers/resources/nodes/syncer.go

```diff
@@ -80,12 +82,62 @@ func (s *nodeSyncer) ModifyController(ctx *synccontext.RegisterContext, builder
 	}()
 	podCache.WaitForCacheSync(ctx.Context)
 	s.podCache = podCache
-	return builder, nil
+	return modifyController(ctx, s.nodeServiceProvider, builder)
+}
+
+func modifyController(ctx *synccontext.RegisterContext, nodeService nodeservice.NodeServiceProvider, builder *builder.Builder) (*builder.Builder, error) {
+	// start the node service provider
+	go func() {
+		nodeService.Start(ctx.Context)
+	}()
+
+	return builder.Watches(source.NewKindWithCache(&corev1.Pod{}, ctx.PhysicalManager.GetCache()), handler.EnqueueRequestsFromMapFunc(func(object client.Object) []reconcile.Request {
+		pod, ok := object.(*corev1.Pod)
+		if !ok || pod == nil || pod.Namespace != ctx.TargetNamespace || !translate.IsManaged(pod) || pod.Spec.NodeName == "" {
```

Comment:
Nit-pick: the `pod.Namespace != ctx.TargetNamespace` part can be removed because `ctx.PhysicalManager` only watches that particular namespace.